### DISCUSSION TITLE
[Question] ESP32 WROOM 32U devkit suitable replacement

### DISCUSSION THREAD

**Discussion by JMVS** [gregmac/Genmon-ESP32-Serial-Bridge#5] [General]
Hello. First of all let me tell you I've been looking for something like for looong time! Thanks for sharing.
I'm not well versed in ESP32 (never owned one) and I would like to know if you can recommend a replacement for it as I don't seem to find it in Argentina. 

The closest things I was able to find is this: https://articulo.mercadolibre.com.ar/MLA-1148515879-modulo-esp32u-wroom-32u-wifi-bluetooth-ble-dual-core-_JM
![imagen](https://user-images.githubusercontent.com/7751148/186026822-bf5abbbf-34f2-4406-9d1b-c4d4cfa0bd05.png)
But I don't know how I would program it.

There are many like this:
![imagen](https://user-images.githubusercontent.com/7751148/186027257-998a1a95-e6e6-4bf4-8868-761aaa923740.png)
But they seem to be missing the external WiFi antenna connector.

I know my way around RPis and some Arduino but as I mentioned before, I never owned an ESP32 to make an informed decision. If you could help me out, it would be appreciated.

**Comment by gregmac** [comment]
@JMVS that's the same module I used, other than the chip antenna instead of external. Whether it will work or not is a factor of wifi signal strength in your situation. Metal, brick, concrete all seriously reduce or even block the signal -- the generator is metal and my house is brick. 

Pretty much any [esphome-compatible device](https://esphome.io/index.html#devices) is likely to work, so long as you can power it with 5V and the [UART bus pins](https://esphome.io/components/uart.html) are exposed.

Side note: I initially was trying to use an ESP8266 board, but found it was very unreliable when powered via 5V pin, and required 2-3 power cycles before it would boot. Apparently this is a notorious and well known problem with these, caused by a cheap power regulator. The one I had used a chip antenna. When the generator panel or lid was open it worked but had a pretty weak signal (my house is brick). As soon as I closed the generator, the device dropped offline. Those two things are why I went to ESP32 and external antenna.

---

For programming, the script (really, YAML file) I have in this repository is for [esphome.io](https://esphome.io/) which provides the code framework, base functionality and the tools to write to the hardware. Maybe useful:

* [Getting Started with the ESPHome Command Line](https://esphome.io/guides/getting_started_command_line.html)
* [Getting Started with ESPHome and Home Assistant](https://esphome.io/guides/getting_started_hassio.html)

**Reply by JMVS** [reply]
Thanks for your reply. I will have to go for the first module shown as I think I will run into the same signal issues myself. What I need is a Serial-USB converter which I already own for flashing my Sonoff devices. In case anyone wonders this should be the basic minimal connection to upload a program:
![imagen](https://user-images.githubusercontent.com/7751148/187098144-21d24f15-e053-43d2-a498-84b19edb613e.png)
(source: https://techtutorialsx.com/2017/06/05/esp-wroom-32-uploading-a-program-with-arduino-ide/)

NOTE: Not tested yet.
